plugins/modules/azure_rm_diagnosticsetting.py

```diff
@@ -0,0 +1,575 @@
+                - The resource Id for the event hub authorization rule.
+            type: str
+            sample: my_event_hub_authorization_rule_id
+        logs:
+            description:
+                - The list of logs settings.
+            type: list
+            contains:
+                category:
+                    description:
+                        - Name of a Diagnostic Log category for a resource type this setting is applied to.
+                    type: str
+                enabled:
+                    description:
+                        - A value indicating whether this log is enabled.
+                    type: bool
+                retention_policy:
+                    description:
+                        - The retention policy for this category.
+                    type: complex
+                    contains:
+                        enabled:
+                            description:
+                                - A value indicating whether the retention policy is enabled.
+                            type: bool
+                        days:
+                            description:
+                                - The number of days for the retention in days. A value of 0 will retain indefinitely.
+                            type: int
+        metrics:
+            description:
+                - The list of metric settings.
+            type: list
+            contains:
+                time_grain:
+                    description:
+                        - The timegrain of the metric in ISO8601 format.
+                    type: str
+                category:
+                    description:
+                        - Name of a Diagnostic Metric category for a resource type this setting is applied to.
+                    type: str
+                enabled:
+                    description:
+                        - A value indicating whether this metric is enabled.
+                    type: bool
+                retention_policy:
+                    description:
+                        - The retention policy for this category.
+                    type: complex
+                    contains:
+                        enabled:
+                            description:
+                                - A value indicating whether the retention policy is enabled.
+                            type: bool
+                        days:
+                            description:
+                                - The number of days for the retention in days. A value of 0 will retain indefinitely.
+                            type: int
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+import datetime
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+retention_policy_object = dict(
+    enabled=dict(type='bool', default=True, required=True),
```

Comment:
Argument 'enabled' in argument_spec found in logs is marked as required but specifies a default. Arguments with a default should not be marked as required